server/src/test/java/org/elasticsearch/cluster/coordination/UnsafeBootstrapAndDetachCommandIT.java

```diff
@@ -353,10 +359,18 @@ public void testAllMasterEligibleNodesFailedDanglingIndexImport() throws Excepti
         assertThat(response.rows()[0][0], is(1L));
 
         logger.info("--> stop data-only node and detach it from the old cluster");
-        Settings dataNodeDataPathSettings = internalCluster().dataPathSettings(dataNode);
+        Settings dataNodeDataPathSettings = Settings.builder()
+            .put(internalCluster().dataPathSettings(dataNode))
+            .put(AUTO_IMPORT_DANGLING_INDICES_SETTING.getKey(), true)
+            .build();
+        assertBusy(() -> internalCluster().getInstance(GatewayMetaState.class, dataNode).allPendingAsyncStatesWritten());
```

Comment:
this was not a part of original commit, but I noticed that we don't have it and added. It's included in the referenced version - https://github.com/elastic/elasticsearch/commit/b8d73a1e7ee86ea14277d064a84915577f2a92ef#diff-cac8bd105e363c72ce4edd25b10c7a9e795f65588abc6bc95ae3e845f3da3c9eR344